# HG changeset patch
# Parent ae7c288669da36406ab98187340d9550af53af81
xen/makefile: Allow XEN_CHANGESET to be set externally

Build systems which build Xen from a source tarball rather than from a mercurial
tree (such as constructing an SRPM then building an RPM from it) will always end
up having the changeset set as "unavailable".

This change allows a build system to be able to provide changeset information
even when the source is not part of a mercurial repository.

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>

diff -r ae7c288669da xen/Makefile
--- a/xen/Makefile
+++ b/xen/Makefile
@@ -13,6 +13,7 @@ export BASEDIR := $(CURDIR)
 export XEN_ROOT := $(BASEDIR)/..
 
 EFI_MOUNTPOINT ?= /boot/efi
+XEN_CHANGESET  ?= $(shell hg root &> /dev/null && hg parents --template "{date|date} {rev}:{node|short}" || echo "unavailable" )
 
 .PHONY: default
 default: build
@@ -107,7 +108,7 @@ include/xen/compile.h: include/xen/compi
 	    -e 's/@@version@@/$(XEN_VERSION)/g' \
 	    -e 's/@@subversion@@/$(XEN_SUBVERSION)/g' \
 	    -e 's/@@extraversion@@/$(XEN_EXTRAVERSION)/g' \
-	    -e 's!@@changeset@@!$(shell ((hg parents --template "{date|date} {rev}:{node|short}" >/dev/null && hg parents --template "{date|date} {rev}:{node|short}") || echo "unavailable") 2>/dev/null)!g' \
+	    -e 's!@@changeset@@!$(XEN_CHANGESET)!g' \
 	    < include/xen/compile.h.in > $@.new
 	@grep \" .banner >> $@.new
 	@grep -v \" .banner
